MediaPlayer/tests/MediaPlayer-specs.js

```diff
@@ -105,4 +114,200 @@ describe('MediaPlayer', () => {
 			expect(mediaSlider).not.toHaveClass('active');
 		});
 	});
+
+	describe('methods and props', () => {
+		test('should have `audio` media', () => {
+			render(
+				<MediaPlayer data-testid="media-player">
+					{audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />))}
+				</MediaPlayer>
+			);
+
+			const audioMedia = screen.getByTestId('media-player').children.item(0);
+
+			expect (audioMedia).toBeInTheDocument();
+		});
+
+		test('should play next audio media when `Next` button is clicked', () => {
+			const handleNext = jest.fn();
+			// onNext added for code coverage purposes
+			render(
+				<MediaPlayer data-testid="media-player" onNext={handleNext}>
+					{audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />))}
+				</MediaPlayer>
+			);
+			const nextButton = screen.getByLabelText('Next');
+
+			userEvent.click(nextButton);
+			const audioSource = screen.getByTestId('media-player').children.item(0).children.item(0);
+
+			const expected = 'https://sampleswap.org/mp3/artist/78152/HiatusManJBanner_Show-Stopper-160.mp3'; // second mp3 file from audioFiles
+			expect(audioSource).toHaveProperty('src', expected);
+		});
+
+		test('should call `onPlay` when `Play` button is clicked', async () => {
+			const handlePlay = jest.fn();
+			render(
+				<MediaPlayer onPlay={handlePlay}>
+					{audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />))}
+				</MediaPlayer>
+			);
+			const playButton = screen.getByLabelText('Play');
+
+			userEvent.click(playButton);
+
+			await waitFor(() => {
+				expect(handlePlay).toHaveBeenCalled();
+			});
+		});
+
+		test('should play previous audio media when `Previous` button is clicked', () => {
+			const handlePrevious = jest.fn();
+			// `onPrevious` added for code coverage purposes
+			render(
+				<MediaPlayer data-testid="media-player" onPrevious={handlePrevious}>
+					{audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />))}
+				</MediaPlayer>
+			);
+			const audioSource = screen.getByTestId('media-player').children.item(0).children.item(0);
+			const nextButton = screen.getByLabelText('Next');
+			const previousButton = screen.getByLabelText('Previous');
+
+			userEvent.click(nextButton); // go to next audio media - audioFiles[1]
+			userEvent.click(previousButton); // go back to first audio media - audioFiles[0]
+
+			const expected = 'https://sampleswap.org/mp3/artist/254731/BossPlayer_Your-Right-Here-160.mp3';
+
+			expect(audioSource).toHaveProperty('src', expected);
+		});
+
```

Comment:
You can also touch the case when repeat=all and the current played media jumps frpm last->first or first->last
This will increase codecov by another 5% 

		test('should play last audio media in the playlist when `Previous` button is clicked and first media in playlist is playing and `repeat=all`', () => {
			const handlePrevious = jest.fn();
			// `onPrevious` added for code coverage purposes
			render(
				<MediaPlayer data-testid="media-player" onPrevious={handlePrevious}>
					{audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />))}
				</MediaPlayer>
			);

			const previousButton = screen.getByLabelText('Previous');

			const repeatButton = screen.getByLabelText('Repeat');

			userEvent.click(repeatButton);
			userEvent.click(repeatButton);

			userEvent.click(previousButton); // go back to last audio media - audioFiles[5]

			const audioSource = screen.getByTestId('media-player').children.item(0).children.item(0);
			const expected = 'https://sampleswap.org/mp3/artist/47067/DJ-Masque_Dont-Forget-To-Be-Yourself-160.mp3';

			expect(audioSource).toHaveProperty('src', expected);
		});

		test('should restart the playlist when `Next` button is clicked and last media in playlist is playing and `repeat=all`', () => {
			const handleNext = jest.fn();
			// `onNext` added for code coverage purposes
			render(
				<MediaPlayer data-testid="media-player" onNext={handleNext}>
					{audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />))}
				</MediaPlayer>
			);

			const nextButton = screen.getByLabelText('Next');

			const repeatButton = screen.getByLabelText('Repeat');

			userEvent.click(repeatButton);
			userEvent.click(repeatButton);

			userEvent.click(nextButton);
			userEvent.click(nextButton);
			userEvent.click(nextButton);
			userEvent.click(nextButton);
			userEvent.click(nextButton);
			userEvent.click(nextButton); // go back to first audio media - audioFiles[0]

			const audioSource = screen.getByTestId('media-player').children.item(0).children.item(0);
			const expected = 'https://sampleswap.org/mp3/artist/254731/BossPlayer_Your-Right-Here-160.mp3';

			expect(audioSource).toHaveProperty('src', expected);
		});